packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -51,48 +51,52 @@ module.exports = async function watchmanCrawl(
-      const canBeFiltered = !existing || existing.length > 0;
+    await Promise.all(
+      roots.map(async root => {
+        const response = await cmd('watch-project', root);
+        const existing = watchmanRoots.get(response.watch);
+        // A root can only be filtered if it was never seen with a relative_path before
+        const canBeFiltered = !existing || existing.length > 0;
 
-      if (canBeFiltered) {
-        if (response.relative_path) {
-          watchmanRoots.set(
-            response.watch,
-            (existing || []).concat(response.relative_path),
-          );
-        } else {
-          // Make the filter directories an empty array to signal that this root
-          // was already seen and needs to be watched for all files/directories
-          watchmanRoots.set(response.watch, []);
+        if (canBeFiltered) {
+          if (response.relative_path) {
+            watchmanRoots.set(
+              response.watch,
+              (existing || []).concat(response.relative_path),
+            );
+          } else {
+            // Make the filter directories an empty array to signal that this root
+            // was already seen and needs to be watched for all files/directories
+            watchmanRoots.set(response.watch, []);
+          }
         }
-      }
-    }
+      }),
+    );
 
     let shouldReset = false;
     const watchmanFileResults = new Map();
-    for (const [root, directoryFilters] of watchmanRoots) {
-      const expression = Array.from(defaultWatchExpression);
-      if (directoryFilters.length > 0) {
-        expression.push([
-          'anyof',
-          ...directoryFilters.map(dir => ['dirname', dir]),
-        ]);
-      }
-      const fields = ['name', 'exists', 'mtime_ms'];
+    await Promise.all(
+      Array.from(watchmanRoots).map(async ([root, directoryFilters]) => {
+        const expression = Array.from(defaultWatchExpression);
+        if (directoryFilters.length > 0) {
+          expression.push([
+            'anyof',
+            ...directoryFilters.map(dir => ['dirname', dir]),
+          ]);
+        }
+        const fields = ['name', 'exists', 'mtime_ms'];
 
-      const query = clocks[root]
-        ? // Use the `since` generator if we have a clock available
-          {expression, fields, since: clocks[root]}
-        : // Otherwise use the `suffix` generator
-          {expression, fields, suffix: extensions};
+        const query = clocks[root]
+          ? // Use the `since` generator if we have a clock available
+            {expression, fields, since: clocks[root]}
+          : // Otherwise use the `suffix` generator
+            {expression, fields, suffix: extensions};
 
-      const response = await cmd('query', root, query);
-      shouldReset = shouldReset || response.is_fresh_instance;
-      watchmanFileResults.set(root, response);
-    }
+        const response = await cmd('query', root, query);
```

Comment:
Instead of what I did above, I can also just return `[root, cmd('query', root, query)]` here and then do
```
const watchmanFileResults = new Map(await Promise.all(...));
```

That said I still need to loop through responses and determine if `shouldReset` needs to be set to true, which would probably be
```
const watcmanRawResults = await Promise.all(...);
const watchmanFileResults = new Map(watcmanRawResults);
const shouldReset = watcmanRawResults.some(([_root, response]) => response.is_fresh_instance);
```

I'm kind of liking this new version but what do you think?

Reply:
I think this is fine tbh